### PR TITLE
PD: auto-created failover pods should be deleted when failed members are recovered

### LINKED ISSUE #2161: PD pods created due to failover are not deleted after the failed pods recover
## Bug Report

**What version of Kubernetes are you using?**

1.12.8
**What version of TiDB Operator are you using?**

1.1.0-rc.1
**What storage classes exist in the Kubernetes cluster and what are used for PD/TiKV pods?**

local-storage
**What's the status of the TiDB cluster pods?**

Running
**What did you do?**

I have 3 pd pods, then pd-1 failed, after 5 minutes, pd-3 is created, but I only have 3 nodes, so pd-3 is in Pending State, after 5 minutes, pd-4 is created and also in Pending state, then pd-1 is Ready
**What did you expect to see?**
pd-3 and pd-4 are deleted because my 3 pd pods are all ready now.
**What did you see instead?**
pd-3 and pd-4 are in Pending state forever.

### PR DESCRIPTION
**Pull request by cofyc**
### What problem does this PR solve? 
fixes #2161 
### Check List 

Tests 

 - Unit test
 - E2E test
 - Stability test
 - Manual test (add detailed scripts or steps below)
 - No code

Code changes

 - Has Go code change
 - Has CI related scripts change
 - Has Terraform scripts change

Side effects

 - Breaking backward compatibility

Related changes

 - Need to cherry-pick to the release branch
 - Need to update the documentation

### Does this PR introduce a user-facing change?:

```release-note
NONE
```

### CODE REVIEW

**Comment by cofyc** [comment]
test job: https://internal.pingcap.net/idc-jenkins/job/tidb-operator-e2e-gke-ci-master/103/

File: tests/e2e/tidbcluster/stability.go [hunk 1]
```diff
@@ -266,9 +269,6 @@ var _ = ginkgo.Describe("[tidb-operator][Stability]", func() {
 				framework.Skipf("current provider is not supported list %v, skipping", supportedProviders.List())
 			}
 
-			ginkgo.By("Wait for all nodes are schedulable")
-			framework.ExpectNoError(framework.WaitForAllNodesSchedulable(c, framework.TestContext.NodeSchedulableTimeout))
-
```
Comment (cofyc) [inline tests/e2e/tidbcluster/stability.go:271]: this is unnecessary because we will wait for at least 3 schedulable nodes below

**Comment by cofyc** [comment]
@Yisaer @DanielZhangQD PTAL

**Comment by DanielZhangQD** [comment]
Should we add release notes for this fix? @cofyc

**Pull request merged**